Tests for constant propagation of setLength() and setCharAt()

danpoe opened the pull request:
<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

danpoe committed:
Bump java-models-library

danpoe committed:
Add tests for StringBuffer.setCharAt()

danpoe committed:
Add tests for StringBuffer.setLength()

danpoe committed:
Add tests for StringBuilder.setCharAt()

danpoe committed:
Add tests for StringBuilder.setLength()

danpoe committed:
Remove StringBuilder.setLength() entry from the string preprocessing table

Preprocessing is unnecessary now as we have a model for setLength() that uses
CProverString.setLength().

danpoe committed:
Remove StringBuilder.setCharAt() entry from the string preprocessing table

Preprocessing is unnecessary now as we have a model for setCharAt() that uses
CProverString.setCharAt().

romainbrenguier commented:
You will need to load the models on the tests in jbmc-strings/StringBuilderSetCharAt/

danpoe committed:
Adapt existing StringBuilder.setCharAt() test

Load models and adapt test so it cannot be solved via constant propagation.

danpoe committed:
Adapt existing StringBuilder tests

Activate two tests marked FUTURE that can meanwhile be solved by the string
solver. Also adapt tests so they cannot be solved via constant propagation, and
load the models library.

danpoe commented:
@romainbrenguier Ah yes I missed that. I've now loaded the library and also adapted the tests so they can't be solved via constant propagation.

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: b44fd4109e117343732716d9c9b687cc0f5bcde5).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/130957817
